src/hyperlight_host/src/hypervisor/regs/standard_regs.rs

```diff
@@ -0,0 +1,420 @@
+            ),
+            (
+                WHvX64RegisterRcx,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rcx }),
+            ),
+            (
+                WHvX64RegisterRdx,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rdx }),
+            ),
+            (
+                WHvX64RegisterRsi,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rsi }),
+            ),
+            (
+                WHvX64RegisterRdi,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rdi }),
+            ),
+            (
+                WHvX64RegisterRsp,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rsp }),
+            ),
+            (
+                WHvX64RegisterRbp,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rbp }),
+            ),
+            (
+                WHvX64RegisterR8,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r8 }),
+            ),
+            (
+                WHvX64RegisterR9,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r9 }),
+            ),
+            (
+                WHvX64RegisterR10,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r10 }),
+            ),
+            (
+                WHvX64RegisterR11,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r11 }),
+            ),
+            (
+                WHvX64RegisterR12,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r12 }),
+            ),
+            (
+                WHvX64RegisterR13,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r13 }),
+            ),
+            (
+                WHvX64RegisterR14,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r14 }),
+            ),
+            (
+                WHvX64RegisterR15,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.r15 }),
+            ),
+            (
+                WHvX64RegisterRip,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rip }),
+            ),
+            (
+                WHvX64RegisterRflags,
+                Align16(WHV_REGISTER_VALUE { Reg64: regs.rflags }),
+            ),
+        ]
+    }
+}
+
+#[cfg(target_os = "windows")]
+use std::collections::HashSet;
+
+#[cfg(target_os = "windows")]
+use super::{Align16, FromWhpRegisterError};
+
+#[cfg(target_os = "windows")]
+pub(crate) const WHP_REGS_NAMES_LEN: usize = 18;
+#[cfg(target_os = "windows")]
+pub(crate) const WHP_REGS_NAMES: [WHV_REGISTER_NAME; WHP_REGS_NAMES_LEN] = [
+    WHvX64RegisterRax,
+    WHvX64RegisterRbx,
+    WHvX64RegisterRcx,
+    WHvX64RegisterRdx,
+    WHvX64RegisterRsi,
+    WHvX64RegisterRdi,
+    WHvX64RegisterRsp,
+    WHvX64RegisterRbp,
+    WHvX64RegisterR8,
+    WHvX64RegisterR9,
+    WHvX64RegisterR10,
+    WHvX64RegisterR11,
+    WHvX64RegisterR12,
+    WHvX64RegisterR13,
+    WHvX64RegisterR14,
+    WHvX64RegisterR15,
+    WHvX64RegisterRip,
+    WHvX64RegisterRflags,
+];
+
```

Comment:
Nit: might be good to add a compile time check that the length of the array matches WHP_REGS_NAMES_LEN?

```
#[cfg(target_os = "windows")]
const _: () = assert!(WHP_REGS_NAMES.len() == WHP_REGS_NAMES_LEN);
```